### PR TITLE
0.3.0 handshake packet

### PR DESCRIPTION
**Pull request by nshaaban-cPacket**
This will be used in socket later.

### CODE REVIEW

**Comment by 1c3t3a** [comment]
Alright! Could you please rebase so that only the relevant changes are shown?

File: src/engineio/packet.rs [hunk 1]
```diff
@@ -41,6 +43,24 @@ const fn u8_to_packet_id(b: u8) -> Result<PacketId> {
...
+    #[serde(rename = "pingInterval")]
+    pub ping_interval: u64,
+    #[serde(rename = "pingTimeout")]
+    pub ping_timeout: u64,
+}
+
+impl TryInto<HandshakePacket> for Packet {
+    type Error = Error;
+    fn try_into(self) -> Result<HandshakePacket> {
+        Ok(serde_json::from_slice::<HandshakePacket>(self.data[..].as_ref())?)
```
Comment (1c3t3a) [inline src/engineio/packet.rs:60]: I think we don't need a turbofish `::<>`here anymore, type inference will catch this.

**Review by 1c3t3a — CHANGES_REQUESTED** [review]
Looks good, only one thing the rest is fine!

File: src/engineio/packet.rs (hunk 1 above)
Comment (nshaaban-cPacket) [reply src/engineio/packet.rs]: done.

**Review by 1c3t3a — APPROVED** [review]

**Pull request merged**